src/devices/BrickPi3/BrickPi3.cs

```diff
@@ -0,0 +1,948 @@
+                        
+                        // convert back the value to a byte array
+                        // we know the length is 2
+                        return new byte[] { (byte)((results[1] >> 8) & 0xFF), (byte)(results[1] & 0xFF), (byte)((results[0] >> 8) & 0xFF), (byte)(results[0] & 0xFF) };
+                    }
+                    else
+                        throw new SensorError(SensorErrorInvalidData);
+                }
+                else
+                    throw new IOError(IOErrorMessage);
+            }
+            else if ((_sensorType[port_index] == SensorType.EV3ColorColorComponents))
+            {
+                outArray.AddRange(new byte[] { SPIAddress, (byte)message_type, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 });
+                reply = SpiTransferArray(outArray.ToArray());
+                if (reply[3] == 0xA5)
+                {
+                    if ((reply[4] == (int)_sensorType[port_index]) || (reply[5] == (int)SensorState.ValidData))
+
+                        return new byte[] { reply[6], reply[7], reply[8], reply[9], reply[10], reply[11], reply[12], reply[13] };
+                    else
+
+                        throw new SensorError(SensorErrorInvalidData);
+                }
+                else
+                    throw new IOError(IOErrorMessage);
+            }
+            else if (_sensorType[port_index] == SensorType.EV3InfraredSeek)
+
+            {
+                outArray.AddRange(new byte[] { SPIAddress, (byte)message_type, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 });
+
+                reply = SpiTransferArray(outArray.ToArray());
+
+                if (reply[3] == 0xA5)
+                {
+                    if ((reply[4] == (int)_sensorType[port_index]) && (reply[5] == (int)SensorState.ValidData))
+
+                    {
+                        return new byte[] { (reply[6]), (reply[7]), (reply[8]), (reply[9]), (reply[10]), (reply[11]), (reply[12]), (reply[13]) };
+                    }
+                    else
+                        throw new SensorError(SensorErrorInvalidData);
+                }
+                else
+                    throw new IOError(IOErrorMessage);
+            }
+            else if (_sensorType[port_index] == Models.SensorType.EV3InfraredRemote)
+            {
+                outArray.AddRange(new byte[] { SPIAddress, (byte)message_type, 0, 0, 0, 0, 0, 0, 0, 0 });
+                reply = SpiTransferArray(outArray.ToArray());
+                if (reply[3] == 0xA5)
+                {
+                    if ((reply[4] == (int)_sensorType[port_index]) && (reply[5] == (int)SensorState.ValidData))
+                    {
+                        byte[] results = new byte[] { 0, 0, 0, 0 };
+                        for (int r = 0; r < results.Length; r++)
+                        {
+                            var value = reply[6 + r];
+                            if (value == 1)
+                                results[r] = 0b10000;
```

Comment:
optional: might be worth to define the array for those values (or some formula if there is such) and just read the index at `value - 1`

Reply:
there is no real formula for this. I would leave it like that for lisibility reasons. So it is closer to the documentation